docs/design/fabric-v2+/fpc-lifecycle-v2.puml

```diff
@@ -18,119 +15,126 @@ title Foot Box removed
+participant Channel order 70
+note over Channel
+    This box abstracts all other orgs and components.
+    Actions are supposed to be performed by channel participants.
+end note
 end box
 
-entity Orderer
-
+entity Orderer order 80
 
-Admin_Org1 -> Admin_Org1 : package
-note left
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_1//
+note over Admin
+    Admin uses the fpc-peer wrappers, which abstract the Fabric Peer CLI.
 end note
 
-Admin_Org1 -> Peer1_Org1 : install package
-return packageID: mycc_1:hash
-Admin_Org1 -> Peer2_Org1 : install package
-return packageID
+group chaincode package
+Admin -> FPC_Peer1_CLI_Wrapper : lifecycle chaincode package
 note right
-	chaincode package identifier
-	is mycc-1:hash
+    **Chaincode package**
+    as tar.gz with
+    metadata.json = <path:(enclaves_path), type:(cc_lang), label:(cc_label), sgx_mode: (sim/hw)>
+    code.tar.gz : compressed file with enclave binary and mrenclave
 end note
-
-Admin_Org2 -> Admin_Org2 : package
-note left
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_1//
+return
+end /' group chaincode package '/
+
+group install
+Admin -> FPC_Peer1_CLI_Wrapper ++: lifecycle chaincode install
+FPC_Peer1_CLI_Wrapper -> Peer1_CLI ++: lifecycle chaincode install
+return packageId
+note over FPC_Peer1_CLI_Wrapper
+    The following bookking operations help remember (across lifecycle calls)
+    that a package (id) corresponds to a FPC chaincode. Similarly, the
+    approve will later store data for an FPC chaincode installed locally,
+    and the commit checks for data of a locally-approved FPC chaincode.
+    Also, it should be noted that
+    - the install lifecycle op is peer-specific, while
+    - the approveformyorg is org-specific, to be run on at most one org's peer
+    - the and commit op is expected to run on a single peer
+    Hence, the current bookeeping ops expect the approve and commit ops to
+    run on peers where the FPC chaincode has been previously installed.
 end note
-
-Admin_Org2 -> Peer1_Org2 : install package
-return packageID: mycc_1:hash
-Admin_Org2 -> Peer2_Org2 : install package
-return packageID
-
-Admin_Org3 -> Admin_Org3 : package
-note left #FFAAAA
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_XXX//
+FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : retrieve cc_language from package
+FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : if cc_language==fpc then store packageId
+return packageId /' Admin -> FPC_Peer1_CLI_Wrapper '/
+end /' group install '/
+
+ref over Admin, FPC_Peer2_CLI_Wrapper, Peer2_CLI
+    install chaincode on Peer2
+end /' ref '/
+
+group local approve
+Admin -> FPC_Peer1_CLI_Wrapper ++: lifecycle chaincode approveformyorg
+FPC_Peer1_CLI_Wrapper -> Peer1_CLI ++: lifecycle chaincode approveformyorg
+note right
+    **Chaincode definition**
+    Name: cc_name, Version: mrenclave,
+    Sequence: sequence_number, Endorsement Policy: (2of3),
```

Comment:
Iff we want protocols to reflect MVP, then endorsement policy cannot be 2of3 but would have to be Org1.peer? But hopefully this would be a policy which we could eventually support so i don't have a fundamental reason to keep it, but maybe worth adding a note that in MVP this would be on a designated peer endorsement policy

BTW: how the designated endorsement policy is represented in the fabric EP language is implicitly part of #358 and we had a corresponding issue in the gdoc mvp workplan, but not yet reflected in github issues. For now i just added a corresponding reference to #358 ...

Reply:
f1fb20ff656811c72403b872681a17b94303cd39